File: OOPS/Inheritance/Hierarchical_Inheritance.java
import java.util.*;
class Alpha{
int a,b;
Scanner sc = new Scanner(System.in);
void input(){
    System.out.println("Enter the 2 numbers:");
    a = sc.nextInt();
    b = sc.nextInt();
}
}
class Beta extends Alpha{
void great(){
    if(a>b)
    System.out.println(a+" is greater.");
    else
    System.out.println(b+" is greater.");
}
}
class Charlie extends Alpha{
void add(){
    System.out.println("The addition of "+a+" + "+b+" = "+(a+b));
}
}
class Hierarchical_Inheritance {
    public static void main(String args[]){
        Beta aa = new Beta();
        Charlie bb = new Charlie();
        aa.input();
        aa.great();
        bb.input();
        bb.add();
    }
}

File: OOPS/hierarchical_inheritance.py
class A:
    a = 10
    b = 20
class B(A):
    def add(self):
        print("The addition of 2 numbers =",(self.a+self.b))
class C(A):
    def multiply(self):
        print("The multiplication of 2 numbers =",(self.a*self.b))
bb = B()
cc = C()
bb.add()
cc.multiply()